lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php

```diff
@@ -62,17 +62,33 @@ public function setReferenceRepository(ReferenceRepository $referenceRepository)
     }
 
     /** @inheritDoc */
-    public function execute(array $fixtures, $append = false)
+    public function execute(array $fixtures, $append = false /* bool $singleTransaction = false */)
     {
-        $executor = $this;
-        $this->em->transactional(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+        $singleTransaction = func_get_args()[2] ?? true;
+
+        if ($singleTransaction) {
+            $executor = $this;
+            $this->em->transactional(static function (EntityManagerInterface $em) use ($executor, $fixtures, $append) {
+                if ($append === false) {
+                    $executor->purge();
+                }
+
+                foreach ($fixtures as $fixture) {
+                    $executor->load($em, $fixture);
+                }
+            });
+        } else {
             if ($append === false) {
-                $executor->purge();
+                $this->em->beginTransaction();
```

Comment:
use `$this->em->getConnection()->transactional()` instead. this will ensure handling properly the rollback on exceptions.

Reply:
So now the code is using `$this->em->wrapInTransaction()`...